apis/sgv2-quarkus-common/src/test/java/io/stargate/sgv2/api/common/properties/datastore/configuration/BridgeDataStorePropertiesRetryTest.java

```diff
@@ -0,0 +1,109 @@
+package io.stargate.sgv2.api.common.properties.datastore.configuration;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import io.grpc.Status;
+import io.grpc.StatusRuntimeException;
+import io.grpc.stub.StreamObserver;
+import io.quarkus.grpc.GrpcClient;
+import io.quarkus.test.junit.QuarkusTest;
+import io.stargate.bridge.proto.Schema;
+import io.stargate.bridge.proto.StargateBridgeGrpc;
+import io.stargate.sgv2.api.common.BridgeTest;
+import io.stargate.sgv2.api.common.config.DataStoreConfig;
+import io.stargate.sgv2.api.common.properties.datastore.DataStoreProperties;
+import java.util.concurrent.atomic.AtomicInteger;
+import javax.inject.Inject;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+
+/**
+ * Tests for verifying retry logic of {@link DataStorePropertiesConfiguration#configuration} call.
+ *
+ * <p>NOTE: properties of {@code @Retry} annotation are overridden to both reduce delay between
+ * calls AND to lower max-retry to 2 calls. See "./src/test/resources/application.yaml" for details.
+ * Override logic by Quarkus/SmallRye is explained <a
+ * href="https://quarkus.io/guides/smallrye-fault-tolerance">here</a>.
+ */
+@QuarkusTest
+public class BridgeDataStorePropertiesRetryTest extends BridgeTest {
```

Comment:
nit: Oh man understanding this test was so hard for me.. This is why I try not to have de-duplication in the test code.. It's almost impossible to read the tests and go through the code with understanding this easily.. Just imagine how this works:

1. Start at the test case -> `dataStoreWithNoRetriesOk() `
2. OK, just delegates to the method, `dataStoreWithNCalls()`, ok that's at the bottom of the class, navigate, understand parameters, etc
3. And now this calls another method `mockGetSupportedFeaturesCall()` which is now top of the class :exploding_head:

Clear `Given -> When -> Then` structure is way better.. Can you at least kill the `mockGetSupportedFeaturesCall` and push all code to the `dataStoreWithNCalls`, there's really no reason to split this up..

Reply:
Ok, I am sorry I caused you headache :)
I did go through couple of rounds of refactoring, limited by this being bit new code to me (wrt mocking). I am bit surprised that it is so hard to follow, given that amount of code is not super high, but YMMV. But I am sure I can improve ordering, structure based on comments above.

Reply:
@ivansenic Ok here goes again. I hope it's bit more readable now.